models/playlist.model.js

```diff
@@ -0,0 +1,17 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const playlistSchema = new Schema(
+	{
+		userId: { type: Schema.Types.ObjectId, ref: "User" },
+		playlists: [
+			{
+				playlistName: { type: String, required: true },
+				videos: [{ type: Schema.Types.ObjectId, ref: "Video" }],
+			},
+		],
+	},
```

Comment:
Instead of having a playlist schema with an array of playlists 
I would suggest each playlist contain a userId and videos array, So every time a playlist is created it gets a new Id and can be pushed to playlists array in User schema , all the operations get simpler

Reply:
Vinay, this will create only one playlist array for each user.